test/helpers.h

```diff
@@ -119,4 +119,14 @@ class PreprocessorHelper
 #define DINIT_NOEXCEPT
 #endif
 
+// Default construct object to avoid bug in clang
+struct default_
```

Comment:
Maybe call it `make_default_obj` instead as the current name is quite obscure.

Also please add the error message to the comment.

Reply:
Done